pennylane/templates/subroutines/prepselprep.py

```diff
@@ -91,19 +92,14 @@ def resource_params(self):
 
     def __init__(self, lcu, control=None, id=None):
 
-        coeffs, ops = lcu.terms()
         control = Wires(control)
-        self.hyperparameters["lcu"] = LinearCombination(coeffs, ops)
-        self.hyperparameters["coeffs"] = coeffs
-        self.hyperparameters["ops"] = ops
+        self.hyperparameters["lcu"] = lcu
         self.hyperparameters["control"] = control
+        target_wires = lcu.wires
 
-        if any(
-            control_wire in Wires.all_wires([op.wires for op in ops]) for control_wire in control
-        ):
+        if any(control_wire in target_wires for control_wire in control):
             raise ValueError("Control wires should be different from operation wires.")
 
-        target_wires = Wires.all_wires([op.wires for op in ops])
```

Comment:
Just store the input `lcu` instead of its `ops` and `coeffs` and a newly created `LinearCombination` instance.
Also, take the `target_wires` directly from the `lcu` instead of computing them from all its constituents.